lib/ruby_ast_gen/node_handling.rb

```diff
@@ -151,20 +154,27 @@ def self.add_node_properties(node_type, base_map, file_path)
       base_map[:body] = children[1]
     when :return, :regopt, *REFS, :redo
       base_map[:value] = children[0] if children[0]
-    when *CONTROL_KW, *FORWARD_ARGUMENTS, :zsuper
+    when *CONTROL_KW 
+      base_map[:arguments] = children[0] if children[0]
+    when *FORWARD_ARGUMENTS, :retry, :zsuper, :match_nil_pattern
       # refer to :type
-
     when *QUAL_ACCESS
       base_map[:base] = children[0]
       base_map[:lhs] = children[1]
       base_map[:rhs] = children[2]
+    when :op_asgn
+      base_map[:lhs] = children[0]
+      base_map[:op] = children[1]
+      base_map[:rhs] = children[2]
     when *ASSIGNMENTS
       base_map[:lhs] = children[0]
       base_map[:rhs] = children[1] if children[1]
     when *BIN_OP
       base_map[:lhs] = children[0]
       base_map[:rhs] = children[1]
-    when *SINGLETONS, *LITERALS, *ARGUMENTS, *ACCESS
+    when *SINGLETONS
+      base_map[:value] = node_type 
```

Comment:
Just double check this. When we have a singleton like `true`, the value is always null which feels wrong to me. Here I set the value to the node_type, so in this case it would be `value = true`, but I can also remove this and just not have a value field. Whichever you feel models the AST better

Reply:
Yeah good to have some value rather than hold `nil` for the sake of it